Hymns/Display/DisplayHymnBottomBarViewModel.swift

```diff
@@ -54,18 +56,22 @@ class DisplayHymnBottomBarViewModel: ObservableObject {
 
                     buttons.append(.fontSize)
 
-                    let languages = self.convertToSongResults(hymn.languages)
-                    if !languages.isEmpty {
-                        buttons.append(.languages(languages))
-                    }
-
                     let mp3Path = hymn.music?.data.first(where: { datum -> Bool in
                         datum.value == DatumValue.mp3.rawValue
                     })?.path
                     if let mp3Url = mp3Path.flatMap({ path -> URL? in
                         HymnalNet.url(path: path)
                     }), self.systemUtil.isNetworkAvailable() {
-                        buttons.append(.musicPlayback(AudioPlayerViewModel(url: mp3Url)))
+                        self.storedMP3Path = mp3Url
+                    }
+
+                    let languages = self.convertToSongResults(hymn.languages)
+                    if !languages.isEmpty {
+                        buttons.append(.languages(languages))
+                    }
+
+                    if let mp3URL = self.storedMP3Path, self.systemUtil.isNetworkAvailable() {
+                        buttons.append(.musicPlayback(AudioPlayerViewModel(url: mp3URL)))
```

Comment:
I think you need to at least first check if the current song has a mp3, and if it does, use that. There are certain songs where the Chinese version has a different tune than the English version. So at most we should use this as a fallback